types/lodash/lodash-tests.ts

```diff
@@ -7389,3 +7407,152 @@ _.templateSettings; // $ExpectType TemplateSettings
     _.chain("").stubFalse(); // $ExpectType LoDashExplicitWrapper<false>
     fp.stubFalse(); // $ExpectType false
 }
+
+/********************
+ * TypeScript Utils *
+ ********************/
+
+// GetIndexedFields: indexing arrays
+{
+    type A1 = Array<'OK'>;
+    type A2 = Array<'OK'> & { x: 'OK' };
+
+    type A = GetIndexedField<A1, number>; // $ExpectType "OK"
+    type B = GetIndexedField<A1, 1>; // $ExpectType "OK"
+    type C = GetIndexedField<A1, '1'>; // $ExpectType "OK"
+    type D = GetIndexedField<A1, 'length'>; // $ExpectType number
+    type E = GetIndexedField<A2, 'x'>; // $ExpectType "OK"
+    type G = GetIndexedField<A1, 'x'>; // $ExpectType undefined
+}
+
+// GetIndexedField: indexing objects
+{
+    const sym = Symbol();
+
+    interface O1 {
+        [sym]: 'sym-OK';
+        [key: string]: 'OK';
+    }
+
+    interface O2 { [key: string]: 'OK' };
+
+    type O3 = Record<string | typeof sym, 'OK'>;
+
+    {   // Trivial cases:
+        type A = GetIndexedField<AbcObject, 'a'>; // $ExpectType number
+        type B = GetIndexedField<AbcObject, 'b'>; // $ExpectType string
+        type D = GetIndexedField<AbcObject, '1'>; // $ExpectType undefined
+        type F = GetIndexedField<AbcObject, number>; // $ExpectType undefined
+        type G = GetIndexedField<O1, typeof sym>; // $ExpectType "sym-OK"
+        type H = GetIndexedField<O1, string>; // $ExpectType "OK"
+        type I = GetIndexedField<O3, typeof sym>; // $ExpectType "OK"
+    }
+
+    {   // Note, TS resolves keyof Ox differently for our test object types:
+        type A = keyof O1 // $ExpectType keyof O1
```

Comment:
These tests aren't very relevant and end up being fairly distracting. I think it would be simpler to just explain in a comment that TS/JS allows using numbers for string indexes.